File: FSUIKit/Classes/FSLabelTextField.m
//
//  FSLabelTextField.m
//  myhome
//
//  Created by fudon on 2017/5/26.
//  Copyright © 2017年 fuhope. All rights reserved.
//

#import "FSLabelTextField.h"

@implementation FSLabelTextField

- (instancetype)initWithFrame:(CGRect)frame text:(NSString *)text textFieldText:(NSString *)tfText placeholder:(NSString *)placeholder{
    self = [super initWithFrame:frame];
    if (self) {
        [self labelTextFieldDesignViews:text textFieldText:tfText placeholder:placeholder];
    }
    return self;
}

- (instancetype)initWithFrame:(CGRect)frame{
    self = [super initWithFrame:frame];
    if (self) {
        [self labelTextFieldDesignViews:nil textFieldText:nil placeholder:@"请输入"];
    }
    return self;
}

- (void)labelTextFieldDesignViews:(NSString *)text textFieldText:(NSString *)tfText placeholder:(NSString *)placeholder{
    CGFloat widthSelf = self.bounds.size.width;
    CGFloat heightSelf = self.bounds.size.height;
    self.backgroundColor = [UIColor whiteColor];
    _label = [[UILabel alloc] initWithFrame:CGRectMake(15, 0, widthSelf / 2, heightSelf)];
    _label.text = text;
    [self addSubview:_label];
    _label.userInteractionEnabled = YES;
    UITapGestureRecognizer *t = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(tapClick)];
    [_label addGestureRecognizer:t];
    
    _textField = [[UITextField alloc] initWithFrame:CGRectMake(widthSelf / 3, 0, widthSelf * 2 / 3 - 15, heightSelf)];
    _textField.placeholder = placeholder;
    _textField.textColor = [UIColor colorWithRed:16/255.0 green:16/255.0 blue:16/255.0 alpha:1.0f];
    _textField.autocorrectionType = UITextAutocorrectionTypeNo;
    _textField.clearButtonMode = UITextFieldViewModeWhileEditing;
    _textField.autocapitalizationType = UITextAutocapitalizationTypeNone;
    _textField.text = tfText;
    _textField.font = [UIFont systemFontOfSize:15];
    _textField.textAlignment = NSTextAlignmentRight;
    [self addSubview:_textField];
}

- (void)tapClick{
    if (self.tapEvent) {
        self.tapEvent(self);
    }
}

/*
 // Only override drawRect: if you perform custom drawing.
 // An empty implementation adversely affects performance during animation.
 - (void)drawRect:(CGRect)rect {
 // Drawing code
 }
 */


@end

File: FSUIKit/Classes/Swift/FSLabelTextField.swift
// FSLabelTextFieldS.swift
// Translated from FSLabelTextField.h/m

import UIKit

public class FSLabelTextField: UIView {
    
    public lazy var label: UILabel = {
        let label = UILabel(frame: CGRect(x: 15, y: 0, width: frame.size.width / 2, height: frame.size.height))
        addSubview(label)
        label.isUserInteractionEnabled = true
        
        let tap = UITapGestureRecognizer(target: self, action: #selector(tapClick))
        label.addGestureRecognizer(tap)
        
        return label
    }()
    
    public lazy var textField: UITextField = {
        let textField = UITextField(frame: CGRect(x: frame.size.width / 3, y: 0, width: frame.size.width * 2 / 3 - 15, height: frame.size.height))
        let rgb = 16 / 255.0
        textField.textColor = UIColor(red: rgb, green: rgb, blue: rgb, alpha: 1.0)
        textField.autocorrectionType = .no
        textField.clearButtonMode = .whileEditing
        textField.autocapitalizationType = .none
        textField.font = UIFont.systemFont(ofSize: 15)
        textField.textAlignment = .right
        addSubview(textField)
        
        return textField
    }()
    
    public var tapEvent: ((FSLabelTextField) -> Void)?
    
    public convenience init(frame: CGRect, text: String?, textFieldText: String?, placeholder: String?) {
        self.init(frame: frame)
        labelTextFieldDesignViews(text: text, textFieldText: textFieldText, placeholder: placeholder ?? "请输入")
    }
    
    public override init(frame: CGRect) {
        super.init(frame: frame)
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
        labelTextFieldDesignViews(text: nil, textFieldText: nil, placeholder: "请输入")
    }
    
    private func labelTextFieldDesignViews(text: String?, textFieldText: String?, placeholder: String) {
        
        let widthSelf = bounds.width
        backgroundColor = .white
        
        label.text = text
        
        textField.text = textFieldText
        textField.placeholder = placeholder
    }
    
    @objc private func tapClick() {
        tapEvent?(self)
    }
}
